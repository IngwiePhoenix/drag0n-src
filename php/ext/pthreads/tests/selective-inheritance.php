<?php
function TestFunction(){
	return __FUNCTION__;
}

define ("TEST_CONSTANT", true);

class TestClass {}

class TestThread extends Thread {
	public function run() { 
		var_dump(function_exists("TestFunction"));
		var_dump(defined("TEST_CONSTANT"));
		var_dump(class_exists("TestClass")); 
	}
}

$thread = new TestThread();
$thread->start();
$thread->join();

$thread = new TestThread();
$thread->start(PTHREADS_INHERIT_NONE);
$thread->join();

$thread = new TestThread();
$thread->start(PTHREADS_INHERIT_FUNCTIONS | PTHREADS_INHERIT_CLASSES);
$thread->join();
?>
