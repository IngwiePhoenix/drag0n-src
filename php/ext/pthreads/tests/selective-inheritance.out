Warning: Zend OPcache can't be temporary enabled (it may be only disabled till the end of request) in Unknown on line 0
bool(true)
bool(true)
bool(true)
bool(false)
bool(false)
bool(false)
bool(true)
bool(false)
bool(true)